completion/available/cargo.completion.bash

```diff
@@ -0,0 +1,259 @@
+        local opt__verify_project="$opt_common $opt_mani $opt_lock"
+        local opt__version="$opt_common $opt_lock"
+        local opt__yank="$opt_common $opt_lock --vers --undo --index --token --registry"
+        local opt__libtest="--help --include-ignored --ignored --test --bench --list --logfile --nocapture --test-threads --skip -q --quiet --exact --color --format"
+
+        if [[ $cword -gt $dd_i ]]; then
+            # Completion after -- separator.
+            if [[ "${cmd}" = @(test|bench) ]]; then
+                COMPREPLY=($(compgen -W "${opt__libtest}" -- "$cur"))
+            else
+                # Fallback to filename completion, useful with `cargo run`.
+                _filedir
+            fi
+        elif [[ $cword -le $cmd_i ]]; then
+            # Completion before or at the command.
+            if [[ "$cur" == -* ]]; then
+                COMPREPLY=($(compgen -W "${opt___nocmd}" -- "$cur"))
+            elif [[ "$cur" == +* ]]; then
+                COMPREPLY=($(compgen -W "$(_toolchains)" -- "$cur"))
+            else
+                COMPREPLY=($(compgen -W "$__cargo_commands" -- "$cur"))
+            fi
+        else
+            case "${prev}" in
+            --vcs)
+                COMPREPLY=($(compgen -W "$vcs" -- "$cur"))
+                ;;
+            --color)
+                COMPREPLY=($(compgen -W "$color" -- "$cur"))
+                ;;
+            --message-format)
+                COMPREPLY=($(compgen -W "$msg_format" -- "$cur"))
+                ;;
+            --manifest-path)
+                _filedir toml
+                ;;
+            --bin)
+                COMPREPLY=($(compgen -W "$(_bin_names)" -- "$cur"))
+                ;;
+            --test)
+                COMPREPLY=($(compgen -W "$(_test_names)" -- "$cur"))
+                ;;
+            --bench)
+                COMPREPLY=($(compgen -W "$(_benchmark_names)" -- "$cur"))
+                ;;
+            --example)
+                COMPREPLY=($(compgen -W "$(_get_examples)" -- "$cur"))
+                ;;
+            --target)
+                COMPREPLY=($(compgen -W "$(_get_targets)" -- "$cur"))
+                ;;
+            --target-dir)
+                _filedir -d
+                ;;
+            help)
+                COMPREPLY=($(compgen -W "$__cargo_commands" -- "$cur"))
+                ;;
+            *)
+                local opt_var=opt__${cmd//-/_}
+                if [[ -z "${!opt_var}" ]]; then
+                    # Fallback to filename completion.
+                    _filedir
+                else
+                    COMPREPLY=($(compgen -W "${!opt_var}" -- "$cur"))
+                fi
+                ;;
+            esac
+        fi
+
+        # compopt does not work in bash version 3
+
+        return 0
+    } &&
+        complete -F _cargo cargo
+
+__cargo_commands=$(cargo --list 2>/dev/null | awk 'NR>1 {print $1}')
+
+_locate_manifest() {
```

Comment:
These bash functions are set globally, so they should share a common prefix to prevent collisions with other plugins.

I suggest **two** underscores, followed by the binary being completed, followed by the function name. E.g. `__cargo_local_manifest`.

This convention should be applied to all functions.